src/monet.js

```diff
@@ -446,6 +446,9 @@
         headMaybe: function () {
             return this.isNil ? None() : Some(this.head_)
         },
+        lookup: function (i) {
+            return this.isNil ? None() : Maybe.fromUndefined(this.toArray()[i])
```

Comment:
1. for performance issues I'd recommend checking if `i` is not equal or higher than length
2. `fromUndefined` will throw if value is `null`
![Screenshot 2021-03-03 at 8 59 10 am](https://user-images.githubusercontent.com/1754948/109780362-db888900-7bfe-11eb-99b7-6067f015c537.png)

Reply:
> for performance issues I'd recommend checking if i is not equal or higher than length
Yea I guess that makes sense 👍 

> fromUndefined will throw if value is null
Ah got it, thanks